homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,254 @@
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY |
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_SWING_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Daikin HVAC platform."""
+    customize = config.get(CONF_CUSTOMIZE, {})
+
+    devices = []
+
+    name = config.get(CONF_NAME, None)
+
+    if discovery_info is not None:
+        ip_address = discovery_info['ip']
+        devices.append(setup_hvac(ip_address, name, customize))
+    else:
+        ip_address = config.get(CONF_IP_ADDRESS, None)
+        devices.append(setup_hvac(ip_address, name, customize))
+
+    add_devices(devices)
+
+
+def setup_hvac(ip_address, name, customize):
+    if ip_address is None:
+        _LOGGER.error("Missing required configuration items %s",
+                      CONF_IP_ADDRESS)
+        return False
+
+    import pydaikin.appliance as appliance
+    device = appliance.Appliance(ip_address)
+
+    if name is None:
+        name = device.values['name']
+
+    return DaikinHVAC(device, name, customize)
+
+
+class DaikinHVAC(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, device, name, customize):
+        import pydaikin.appliance as appliance
+
+        """Initialize the HVAC."""
+        self._name = name
+        self._device = device
+        self._customize = customize
+
+        self._current_temperature = self.settings('htemp', True)
+        self._operation_list = list(DAIKIN_OPERATION_LIST.values())
+        self._fan_list = appliance.daikin_values('f_rate')
+        self._swing_list = appliance.daikin_values('f_dir')
+
+        self._target_temperature = None
+        self._current_humidity = None
+        self._target_humidity = None
+        self._current_fan_mode = None
+        self._current_swing_mode = None
+
+    @property
+    def unique_id(self):
+        """Return the ID of this AC."""
+        return "{}.{}".format(self.__class__, self._device.ip)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def should_poll(self):
+        """Polling needed for a demo climate device."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        self._current_temperature = self.settings('htemp', True)
+
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
```

Comment:
Does the call to `self.settings()` do I/O or just retrieve cached values from the library? I don't think so, just making sure.

Also, since you just return whatever the library returns: Do you really need the local `self._target_temperature`? (Also applies in other places…)

Reply:
Indeed I'm retrieving the values from the library.
Nothing from this file is doing I/O operations. All I do is retrieve the values from the library :)